### ISSUE TITLE
Query Model Structure.

### ISSUE THREAD

**Issue by hemantjadon** [fossasia/loklak_search#67]
There should be model/class for query string processing, which can contain various filters and aspects of a search, `mentions`, `hashtags`, and most importantly `time and location` based searches, along with client side filtering like, `videos/images/news`
- There should be direct linking of query string (from url) to a query object on component initialization.
- On subsequent changes in query the object is modified and various filters can be applied on the object.
- This will decouple the query string logic into a more robust object whose properties can be changed irrespective of the component.
  
  Eg. We want to implement the time restricted search in #43, this will be much easier to normalise if there is only one query object with property say `since` and `until` if these properties are set irrespective of which component sets it, the query will be performed accordingly.
- Basically There can be single query object in the application whose properties varies according to the applied filters and narrowing of search results. These can be varied by different components and query will be performed on cumulative changes rather than each change requesting the `search service` to fetch the results.
- Using this approach debouncing and defering of searches can be performed as there will be single requesting source.

Please provide inputs and suggestions :)

**Comment by mariobehling** [comment]
Sounds good. Please proceed.

**Comment by uttu357** [comment]
@hemantjadon Can i work on this along with implementing the design of time and location filter based search?

**Comment by hemantjadon** [comment]
@uttu357 I have written the structure already and that part is almost complete. I will submit the PR in 1-2 days then we can work on this.

**Comment by uttu357** [comment]
@hemantjadon sure! :+1:

**Comment by meets2tarun** [comment]
@hemantjadon  Are you still working on this ?
I think your PR will solve https://github.com/fossasia/loklak_search/issues/43

**Comment by hemantjadon** [comment]
@meets2tarun yes I am.

**Comment by uttu357** [comment]
@hemantjadon Are you currently doing this?

**Comment by hemantjadon** [comment]
Fixed now
